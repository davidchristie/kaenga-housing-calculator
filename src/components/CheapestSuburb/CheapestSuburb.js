import React, { Component, PropTypes } from 'react'
import { Well } from 'react-bootstrap'

class CheapestSuburb extends Component {
  render () {
    const style = this.props.visible ? {} : {display: 'none'}
    return (
      <Well style={style}>
        <h1>Cheapest suburb</h1>
        <p>
          Name: <strong>{this.props.name}</strong>
        </p>
      </Well>
    )
  }
}

CheapestSuburb.propTypes = {
  price: PropTypes.number,
  name: PropTypes.string
}

export default CheapestSuburb
