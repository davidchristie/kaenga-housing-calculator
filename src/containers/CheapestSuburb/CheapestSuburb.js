import { connect } from 'react-redux'

import CheapestSuburb from '../../components/CheapestSuburb'
import { getSelectedSuburbs } from '../../selectors/suburbs'

export default connect(
  state => {
    const selected = getSelectedSuburbs(state)
    const prices = selected.map(suburb => {
      return suburb.price
    })
    const min = Math.min(...prices)
    const cheapest = selected.find(suburb => {
      return suburb.price === min
    })
    return {
      ...cheapest,
      visible: cheapest !== undefined
    }
  }
)(CheapestSuburb)
