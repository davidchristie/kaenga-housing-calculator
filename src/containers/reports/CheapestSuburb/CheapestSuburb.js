import { connect } from 'react-redux'

import CheapestSuburb from '../../../components/reports/CheapestSuburb'
import { getCheapestSelectedSuburb } from '../../../selectors/suburbs'

export default connect(
  state => {
    const cheapest = getCheapestSelectedSuburb(state)
    return {
      ...cheapest,
      visible: cheapest !== undefined
    }
  }
)(CheapestSuburb)
