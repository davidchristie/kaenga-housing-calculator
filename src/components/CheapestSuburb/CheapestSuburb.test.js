import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import CheapestSuburb from './CheapestSuburb'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<CheapestSuburb />)).toMatchSnapshot()
})
