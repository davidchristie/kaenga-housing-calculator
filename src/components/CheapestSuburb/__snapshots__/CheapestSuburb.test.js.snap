exports[`test matches snapshot 1`] = `
<Well
  bsClass="well"
  style={
    Object {
      "display": "none",
    }
  }>
  <h1>
    Cheapest suburb
  </h1>
  <p>
    Name: 
    <strong />
  </p>
  <p>
    Average house price: 
    <strong>
      $
    </strong>
  </p>
</Well>
`;
